src/stim/simulators/tableau_simulator.cc

```diff
@@ -715,7 +723,7 @@ void TableauSimulator::Z(const OperationData &target_data) {
     }
 }
 
-simd_bits<MAX_BITWORD_WIDTH> TableauSimulator::sample_circuit(const Circuit &circuit, std::mt19937_64 &rng, int8_t sign_bias) {
+simd_bits<MAX_BITWORD_WIDTH> TableauSimulator::sample_circuit(const Circuit &circuit, const std::mt19937_64 &rng, int8_t sign_bias) {
```

Comment:
Is this supposed to have been changed to const? The semantics of the method are changed pretty drastically by this change. This may have broken some of the fuzz tests so that they only check one case.

This method is clearly a sortof leftover from the old way the code was using RNGs by passing them around. It probably needs to be refactored much more extremely in order to make it work with the new way where the simulators own their RNGs. Either the RNG state needs to be written back into this value as the method returns, or the parameter needs to be refactored to be like a seed or something.

Reply:
Good catch! Writing back RNG state would work for now, but seems a bit ugly and unnecessarily costly (mt19937_64 weighs about 2.5kB, see above). Also, the two functions have very few callsites, so I'd like to just fix them in this PR.

It seems to me the most elegant way to fix this is to make the two functions non-static. This simplifies function signatures and ensures that RNG initialization concerns are addressed in one place (the c'tor of TableauSimulator). The downside is that the caller has to explicitly construct the sim. However, that construction is very simple since the c'tor has only one non-optional arg (the RNG).

(Note that `reference_sample_circuit` remains static since the RNG is irrelevant.)